core/src/main/java/jenkins/widgets/HistoryPageFilter.java

```diff
@@ -101,85 +105,100 @@ public void setSearchString(@Nonnull String searchString) {
     /**
      * Add build items to the History page.
      *
-     * @param items The items to be added. Assumes the list of items are in descending queue ID order i.e. newest first.
+     * @param runItems The items to be added. Assumes the list of items are in descending queue ID order i.e. newest first.
      */
-    public void add(@Nonnull List<T> items) {
-        if (items.isEmpty()) {
+    public void add(@Nonnull Iterable<T> runItems) {
```

Comment:
@dennisjlee Is there a way to keep the old method around for compatibility, annotate it `@Deprecated` and have it do something sensible, e.g. calling the new method? If so, that should be the preferred approach.